### PR TITLE
EZP-29276 - As a QA I want to cover scenarios of testing media, binar…

### PR DESCRIPTION
**Pull request by m-tyrala**
…y and image fields in content management

| Question      | Answer
| ------------- | ---
| Tickets       | [EZP-29276](https://jira.ez.no/browse/EZP-29276)
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

As a QA
I want to cover with tests ezmedia, ezbinary and ezimage fields within the area of basic management of content edition and creation
To increase the detection of errors in this area

#### Checklist:
- [ ] Coding standards (`$ composer fix-cs`)
- [ ] Ready for Code Review

### CODE REVIEW

File: behat_suites.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # This file is meant to be imported from ezplatform's behat.yml.dist.
 # All path are relative to the root ezplatform directory.
+default:
+    extensions:
+        Behat\MinkExtension:
+            files_path: '%paths.base%/vendor/ezsystems/ezplatform-admin-ui/src/lib/Behat/TestFiles/'
```
Comment (micszo) [inline behat_suites.yml]: In the bundle not in meta repo after all - what was the final argument? 🙂

File: src/lib/Behat/TestFiles/binary1.txt [hunk 2]
```diff
@@ -0,0 +1,85 @@
+Feature: Content fields setting and editing
```
Comment (micszo) [inline src/lib/Behat/TestFiles/binary1.txt]: Is a text file a good enough binary file? 🙂  What is the objective here?

Comment (m-tyrala) [reply src/lib/Behat/TestFiles/binary1.txt]: Mainly the size of the file - `txt` is not an image, neither a video and at the same time it's easy to prepare a small text file, unlike some encoded formats like `exe` or `dmg`.

**Comment by micszo** [comment]
@m-tyrala could you rebase after https://github.com/ezsystems/ezplatform-admin-ui/pull/567 was merged? 🙂

File: src/lib/Behat/Helper/UtilityContext.php [hunk 3]
```diff
@@ -271,4 +271,25 @@ public function waitUntilElementDisappears(string $cssSelector, int $timeout): v
...
+    /**
+     * Uploads file from location stored in 'files_path' to the disc on remote browser machine.
+     *
+     * @param string $localFileName
+     *
+     * @return string
+     */
+    public function uploadFileToRemoteSpace(string $localFileName): string
+    {
+        $localFile = sprintf('%s%s', $this->getMinkParameter('files_path'), $localFileName);
```
Comment (webhdx) [inline src/lib/Behat/Helper/UtilityContext.php]: If we need a file for testing purposes why can't you simply create a `zip` archive in the repository? I'm mostly concerned about the performance of this.

File: src/lib/Behat/PageElement/Fields/File.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUi\Behat\PageElement\Fields;
```
Comment (webhdx) [inline src/lib/Behat/PageElement/Fields/File.php]: Please add `declare(strict_types=1);`. Tests should follow the same coding standards as the source code. I suspect this is an issue among all tests related files but maybe you can align it slowly in future PRs.

File: src/lib/Behat/PageElement/Fields/File.php [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+use PHPUnit\Framework\Assert;
+
+class File extends EzFieldElement
+{
+    /** @var string Name by which Element is recognised */
+    public const ELEMENT_NAME = 'File';
+
+    public function __construct(UtilityContext $context, string $locator, string $label)
+    {
+        parent::__construct($context, $locator, $label);
```
Comment (webhdx) [inline src/lib/Behat/PageElement/Fields/File.php]: New line below `parent::__construct()`.

**Comment by m-tyrala** [comment]
@webhdx I'll take into account this tips in the next PRs. However, I fixed the "zip" issue - I guess sometimes it is most difficult to come up with the simplest solutions 😁

**Pull request merged**